src/GCDWheel.cs

```diff
@@ -61,18 +71,83 @@ public unsafe class GCDWheel {
                 targetBuffer = DataStore.ClientState.LocalPlayer.TargetObjectId;
             if (addingToQueue) {
                 AddToQueue(act, isWeaponSkill);
+                    queuedAbilityName = GetAbilityName(actionID, DataStore.ClientState.LocalPlayer.CastActionType);
             } else {
                 if (isWeaponSkill) {
                     EndCurrentGCD(TotalGCD);
                     //Store GCD in a variable in order to cache it when it goes back to 0
                     TotalGCD = act->TotalGCD;
                     AddWeaponSkill(act);
+                        queuedAbilityName = GetAbilityName(actionID, DataStore.ClientState.LocalPlayer.CastActionType);
                 } else if (!executingQueued) {
                     ogcds[act->ElapsedGCD] = new(act->AnimationLock, false);
                 }
             }
         }
 
+        //probably should find a way to do this from DataStore so we aren't passing the world
+        //into GCDWheel
+        private string GetAbilityName(uint actionID, byte actionType) {
+            var lumina = dataManager;
+
+            switch (actionType) {
+                // Seem to need case 0 here for follow up casts for short spells (gcdTime > castTime).
+                case 0:
+                case 1:
+                    var ability = lumina?.GetExcelSheet<Lumina.Excel.GeneratedSheets.Action>()?.GetRow(actionID);
+                    return ability?.Name ?? "Unknown Ability";
+
+                case 2:
+                    var item = lumina?.GetExcelSheet<Lumina.Excel.GeneratedSheets.Item>()?.GetRow(actionID);
+                    return item?.Name ?? "Unknown Item";
+
+                case 13:
+                    var mount = lumina?.GetExcelSheet<Lumina.Excel.GeneratedSheets.Mount>()?.GetRow(actionID);
+                    return mount != null ? CapitalizeOutput(mount.Singular) : "Unknown Mount";
+
+                default:
+                    //so, we're not going to talk about this, and I'm going to deny ever doing it.
```

Comment:
I didn't spend any time trying to do this more elegantly because a) we don't need this now that we are getting the game's castbar text directly, and b) I felt like there must be a much better way to do it than to compare the string so I was a garbage implementation anyway.  I left it in since we are still using the case 0 & 1 above to get the queued spell name, and with it all in place we have a quick fallback in case Dalamud ever breaks the ATKStage implementation.

I'll have a look at spiffing it up some.  It's unused at the moment though.